### PR TITLE
Dom bridge lexbor

### PR DESCRIPTION
**Pull request by nielsdos**
TODO:

- [x] more tests
- [x] loadHTML5File
- [x] error reporting
- [x] custom memory allocators
- [x] Windows support
- [x] Hook up getLineNo? Set column too ?
- [x] ~~Recover flag testen en eventueel error reporting adhv dat aanpassen~~ is only sensible for XML, HTML auto-recovers by default
- [x] Measure increase in dom.so size
- [x] Perform some benchmarks of parsing on top X websites?
- [x] Minder lelijke error handling hacks?
- [x] codecov ignore file?
- [x] ~~Hook up external entity loader~~ Actually, let's keep this for a secondary RFC vote because I don't want to waste time on it if it isn't wanted.
- [x] README.REDIST.BINS
- [x] encoding/decoding
- [x] interaction ->encoding test
- [x] serializer
- [x] edge cases encoding (long documents, euc-jp finish)
- [x] serializer tests met verschillende root types
- [x] ~~ifdefs~~
- [x] escape tests, nbsp tests, ...
- [x] new column handling
- [x] Encoding finalising test?
- [x] Rebase and clean history
- [x] element namespace handling (xhtml, svg, mathml)
- [x] ~~default register xhtml namespace in xpath?~~ NO_DEFAULT_NS migration
- [x] optimize jumping in utf-8 parsing?
- [x] Also apply jumping in Lexbor, but make sure the end pointer ends at the predetermined place
- [x] Put benchmark stuff in gist

Try it out instructions: https://gist.github.com/nielsdos/139099c54ddc4a43bc66f4c0b7ef02fc

### CODE REVIEW

File: ext/dom/lexbor_bridge/lexbor/html/tokenizer.c [hunk 1]
```diff
@@ -91,6 +91,7 @@ lxb_html_tokenizer_init(lxb_html_tokenizer_t *tkz)
 
     tkz->pos = tkz->start;
     tkz->end = tkz->start + LXB_HTML_TKZ_TEMP_SIZE;
+    /* current_offset already initialized by calloc */
```
Comment (nielsdos) [inline ext/dom/lexbor_bridge/lexbor/html/tokenizer.c:94]: TODO: update comment

File: ext/dom/lexbor_bridge/lexbor/html/tokenizer.c [hunk 2]
```diff
@@ -312,7 +315,16 @@ lxb_html_tokenizer_chunk(lxb_html_tokenizer_t *tkz, const lxb_char_t *data,
     tkz->last = end;
 
     while (data < end) {
-        data = tkz->state(tkz, data, end);
+        const lxb_char_t *new_data = tkz->state(tkz, data, end);
+        for (const lxb_char_t *ptr = data; ptr < new_data; ptr++) {
```
Comment (nielsdos) [inline ext/dom/lexbor_bridge/lexbor/html/tokenizer.c:319]: ptr & data can be merged
